pkg/image/api/validation/validation.go

```diff
@@ -185,3 +183,45 @@ func ValidateImageStreamTagUpdate(newIST, oldIST *api.ImageStreamTag) fielderror
 
 	return result
 }
+
+func ValidateImageStreamImport(isi *api.ImageStreamImport) fielderrors.ValidationErrorList {
+	errs := fielderrors.ValidationErrorList{}
+	for i, spec := range isi.Spec.Images {
+		from := spec.From
+		switch from.Kind {
+		case "DockerImage":
+			if spec.To != nil && len(spec.To.Name) == 0 {
+				errs = append(errs, fielderrors.ValidationErrorList{fielderrors.NewFieldInvalid("to.name", spec.To.Name, "the name of the target tag must be specified")}.PrefixIndex(i)...)
+			}
+			if len(spec.From.Name) == 0 {
+				errs = append(errs, fielderrors.ValidationErrorList{fielderrors.NewFieldRequired("from.name")}.PrefixIndex(i)...)
+			} else {
+				if _, err := api.ParseDockerImageReference(spec.From.Name); err != nil {
+					errs = append(errs, fielderrors.ValidationErrorList{fielderrors.NewFieldInvalid("from.name", spec.From.Name, err.Error())}.PrefixIndex(i)...)
+				}
+			}
+		default:
+			errs = append(errs, fielderrors.ValidationErrorList{fielderrors.NewFieldInvalid("from.kind", from.Kind, "only DockerImage is supported")}.PrefixIndex(i)...)
+		}
+	}
+	errs = errs.Prefix("spec.images")
```

Comment:
It's a bunch more boilerplate that makes the errors harder to read, and if we need to solve it it needs to be solved upstream in field errors (to have a formal way to join the prefixes).